src/cmd/tools/split_shards/main/main.go

```diff
@@ -0,0 +1,337 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	iofs "io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pborman/getopt"
+	"go.uber.org/zap"
+
+	"github.com/m3db/m3/src/dbnode/persist"
+	"github.com/m3db/m3/src/dbnode/persist/fs"
+	"github.com/m3db/m3/src/dbnode/sharding"
+	"github.com/m3db/m3/src/x/ident"
+	xtime "github.com/m3db/m3/src/x/time"
+)
+
+var checkpointPattern = regexp.MustCompile(`/data/(\w+)/([0-9]+)/fileset-([0-9]+)-([0-9]+)-checkpoint.db$`)
+
+func main() {
+	var (
+		optSrcPath       = getopt.StringLong("src-path", 's', "", "Source path [e.g. /temp/lib/m3db/data]")
+		optDstPathPrefix = getopt.StringLong("dst-path", 'd', "", "Destination path prefix [e.g. /var/lib/m3db/data]")
+		optBlockUntil    = getopt.Int64Long("block-until", 'b', 0, "Block Until Time [in nsec]")
+		optShards        = getopt.Uint32Long("src-shards", 'h', 0, "Original (source) number of shards")
+		optFactor        = getopt.IntLong("factor", 'f', 0, "Integer factor to increase the number of shards by")
+	)
+	getopt.Parse()
+
+	rawLogger, err := zap.NewDevelopment()
+	if err != nil {
+		log.Fatalf("unable to create logger: %+v", err)
+	}
+	logger := rawLogger.Sugar()
+
+	if *optSrcPath == "" ||
+		*optDstPathPrefix == "" ||
+		*optSrcPath == *optDstPathPrefix ||
+		*optBlockUntil <= 0 ||
+		*optShards == 0 ||
+		*optFactor <= 0 {
+		getopt.Usage()
+		os.Exit(1)
+	}
+
+	var (
+		srcFilesetLocation = dropDataSuffix(*optSrcPath)
+		dstFilesetLocation = dropDataSuffix(*optDstPathPrefix)
+
+		srcFsOpts = fs.NewOptions().SetFilePathPrefix(srcFilesetLocation)
+		dstFsOpts = fs.NewOptions().SetFilePathPrefix(dstFilesetLocation)
```

Comment:
Hm, think the getopt "e.g. /var/lib/m3db/data" should instead be "e.g. /var/lib/m3db" since the `/data` will be implied when using the data fileset reader.

Reply:
I drop `/data` suffix if present (intentionally, see my reply to https://github.com/m3db/m3/pull/3632/files#r690370535 for reasoning), see `func dropDataSuffix`.